### DISCUSSION TITLE
Gcc-states-latest.osrm.datasource_names not found

### DISCUSSION THREAD

**Discussion by sahil0094** [Project-OSRM/osrm-backend#6819] [Q&A]
# Issue
I am trying to host osrm server within my azure environment using azure container instance for gcc region. I followed the steps mentioned in the article https://blog.vincentcos.tel/getting-started-with-osrm-on-azure-container-instances/. Everything works fine till osrm-partition command. Post executing osrm-customize I can see below two files are missing error 
![IMG_4833](https://github.com/Project-OSRM/osrm-backend/assets/31719914/ccd8b0e5-69d0-493c-88ad-45e97f3f118e)

after osrm-customize I ran the command to run the server post which the container terminates with error below
![image](https://github.com/Project-OSRM/osrm-backend/assets/31719914/4b18532f-045f-4e04-8375-3dbf5e4bce06)

please help me identify the issue

# Steps to reproduce

Please follow the blog mentioned below. I’m doing it for Asia-> gcc region

https://blog.vincentcos.tel/getting-started-with-osrm-on-azure-container-instances/

# Specifications
I’m using azure container instance with 2 CPUs and 4 GB RAM

**Comment by frodrigo** [comment]
Does the contract step finish without error ?

**Comment by sahil0094** [comment]
> Does the contract step finish without error ?

You mean customise? No, it shows error. 2 files missing , check below

![image](https://github.com/Project-OSRM/osrm-backend/assets/31719914/82ed024f-56e6-4095-a6b4-67188ee1f049)

**Comment by frodrigo** [comment]
Yes. The question is, if files are missing, the previous step ends without
errors ?

**Comment by sahil0094** [comment]
> Yes. The question is, if files are missing, the previous step ends without
> 
> errors ?
> 
> 
> 
> El sáb, 25 nov 2023, 19:57, infinity ***@***.***> escribió:
> 
> 
> 
> > Does the contract step finish without error ?
> 
> >
> 
> > You mean customise? No, it shows error. 2 files missing , check below
> 
> >
> 
> > 5FF52A51-A768-4D37-8599-D36B5A669912.jpeg (view on web)
> 
> > <https://github.com/Project-OSRM/osrm-backend/assets/31719914/82ed024f-56e6-4095-a6b4-67188ee1f049>
> 
> >
> 
> > —
> 
> > Reply to this email directly, view it on GitHub
> 
> > <https://github.com/Project-OSRM/osrm-backend/issues/6742#issuecomment-1826398352>,
> 
> > or unsubscribe
> 
> > <https://github.com/notifications/unsubscribe-auth/AANT5DWINY467CN4AKSICSTYGI5QRAVCNFSM6AAAAAA7ZBWQKOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMRWGM4TQMZVGI>
> 
> > .
> 
> > You are receiving this because you commented.Message ID:
> 
> > ***@***.***>
> 
> >
> 
> 

Yes, the step before this is partition and there it ends without any error

**Comment by sahil0094** [comment]
Any suggestions?

**Comment by frodrigo** [comment]
Are you sure it finish well ? Have you checked the complete logs, the result status $?, the disk space...

**Comment by sahil0094** [comment]
> Are you sure it finish well ? Have you checked the complete logs, the result status $?, the disk space...

Yes, no error in previous step

**Comment by danpat** [comment]
@sahil0094 The missing files indicate that the previous step did _not_ succeed.  Can you paste the output of the `osrm-extract` step?

https://blog.vincentcos.tel/getting-started-with-osrm-on-azure-container-instances/#run-osrm-extract

The instructions indicate that you need to confirm that the last line logged is `[info] RAM: peak bytes used: <something>`  which is correct.  If that is _not_ the last line, then your `osrm-extract` command did not complete properly and that is why the files are missing.

**Comment by sahil0094** [comment]
> @sahil0094 The missing files indicate that the previous step did _not_ succeed.  Can you paste the output of the `osrm-extract` step?
> 
> 
> 
> https://blog.vincentcos.tel/getting-started-with-osrm-on-azure-container-instances/#run-osrm-extract
> 
> 
> 
> The instructions indicate that you need to confirm that the last line logged is `[info] RAM: peak bytes used: <something>`  which is correct.  If that is _not_ the last line, then your `osrm-extract` command did not complete properly and that is why the files are missing.

Yes, I did get that logs "[info] RAM: peak bytes used: <something>" post extract step. Let me share 
![image](https://github.com/Project-OSRM/osrm-backend/assets/31719914/bda668b8-7913-4848-ae85-3d7d207371ed)

**Comment by danpat** [comment]
If that worked cleanly, then it seems like you've got some kind of Azure container volume issue - perhaps you're not accessing the same data location, or your container data is ephemeral (e.g. lost when the container shuts down).  I am not familiar enough with persistent storage on Azure to know how to help with that - but make sure you're writing the data (`/usr/csuser/clouddrive`) to a location that will be available to each container invocation.

**Comment by sahil0094** [comment]
I believe container has access to all the files otherwise it would have given error for a lot more files missing. Right now the file that's missing (osrm.datasource_names) is not even getting created in the location whereas all other files are there. Can there be any reason why this particular file is not getting created?
